test/roby_app/ftp_server/test_server.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+require "minitest/autorun"
+
+require "ftpd"
+require "net/ftp"
+
+require "syskit/roby_app/ftp_server"
+
+class FtpServerTest < Minitest::Test
+    # auxiliary functions
+
+    def spawn_server
+        @temp_dir = Ftpd::TempDir.make
+        @server = Syskit::RobyApp::FtpServer::Server.new(@temp_dir, user: "test.user")
+        @certificate = "/home/#{ENV['LOGNAME']}/.local/share/autoproj/gems/ruby/2.5.0/gems/ftpd-2.1.0/insecure-test-cert.pem"
```

Comment:
This assumes the environment is exactly like on your machine. Which is not in CI for instance (the stuff does not necessarily get installed in `/home`

You do need to access the `insecure_certfile_path`. One way would be to:
- create a subclass of the "real" Server, within the test, file that includes InsecureCertificate and uses insecure_certfile_path as certificate by default
- use this subclass in the tsts
- add an accessor to get the server's certificate path